### DISCUSSION TITLE
WS2801 leds

### DISCUSSION THREAD

**Discussion by wou5er** [awawa-dev/HyperSerialPico#46] [Q&A]
Are these supported? 
APA102 had data and clock? 

I'm using the Feather RP2040

Would HyperSerialPico_Spi.uf2 be sufficient,

**Comment by awawa-dev** [comment]
Nobody has tested it. Also may require to change the SPI speed from 10MHz to 1MHz

**Reply by wou5er** [reply]
how can i change the spi,speed do i need to re compile for that or is it in hyperHDR?

**Reply by awawa-dev** [reply]
did you try the current firmware and 10Mb speed doesnt work? you need to compile the firmware online (manual in wiki) after you change this line in your fork https://github.com/awawa-dev/HyperSerialPico/blob/fb91be4f0a5e1e38b9606827eea3bd957c42010a/include/leds.h#L504C23-L504C26

**Reply by wou5er** [reply]
didn't try it yet. had to get some info from you first. 

I flashed the pico so i'll give it a try with stock.

**Reply by wou5er** [reply]
i flashed that firmware and connected everything but i'm not getting the impression it working like it should 

`
[    0.000000] Booting Linux on physical CPU 0x0000000000 [0x410fd034]
[    0.000000] Linux version 6.2.0-odroid-arm64 (tobetter@gmail.com) (gcc-10 (Debian 10.2.1-6) 10.2.1 20210110, GNU ld (GNU Binutils for Debian) 2.35.2) #1 SMP PREEMPT Tue, 31 Jan 2023 07:28:15 +0000
[    0.000000] Machine model: Hardkernel ODROID-N2
[    0.000000] OF: fdt: Reserved memory: failed to reserve memory for node 'secmon@5300000': base 0x0000000005300000, size 32 MiB
[    0.000000] Reserved memory: created CMA memory pool at 0x00000000dd800000, size 256 MiB
[    0.000000] OF: reserved mem: initialized node linux,cma, compatible id shared-dma-pool
[    0.000000] NUMA: No NUMA configuration found
[    0.000000] NUMA: Faking a node at [mem 0x0000000000000000-0x00000000ed7fffff]
[    0.000000] NUMA: NODE_DATA [mem 0xdd01da80-0xdd01ffff]
[    0.000000] Zone ranges:
[    0.000000]   DMA      [mem 0x0000000000000000-0x00000000ed7fffff]
[    0.000000]   DMA32    empty
[    0.000000]   Normal   empty
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000004ffffff]
[    0.000000]   node   0: [mem 0x0000000005000000-0x00000000052fffff]
[    0.000000]   node   0: [mem 0x0000000005300000-0x00000000ed7fffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x00000000ed7fffff]
[    0.000000] On node 0, zone DMA: 10240 pages in unavailable ranges
[    0.000000] psci: probing for conduit method from DT.
[    0.000000] psci: PSCIv1.0 detected in firmware.
[    0.000000] psci: Using standard PSCI v0.2 function IDs
[    0.000000] psci: MIGRATE_INFO_TYPE not supported.
[    0.000000] psci: SMC Calling Convention v1.1
[    0.000000] percpu: Embedded 20 pages/cpu s44392 r8192 d29336 u81920
[    0.000000] pcpu-alloc: s44392 r8192 d29336 u81920 alloc=20*4096
[    0.000000] pcpu-alloc: [0] 0 [0] 1 [0] 2 [0] 3 [0] 4 [0] 5
[    0.000000] Detected VIPT I-cache on CPU0
[    0.000000] CPU features: detected: ARM erratum 845719
[    0.000000] alternatives: applying boot alternatives
[    0.000000] Fallback order for Node 0: 0
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 957600
[    0.000000] Policy zone: DMA
[    0.000000] Kernel command line: root=/dev/mmcblk0p2 rootwait ro quiet console=tty1 clk_ignore_unused fsck.mode=force fsck.repair=yes net.ifnames=0 console=ttyS0,115200n8
[    0.000000] Dentry cache hash table entries: 524288 (order: 10, 4194304 bytes, linear)
[    0.000000] Inode-cache hash table entries: 262144 (order: 9, 2097152 bytes, linear)
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 3514448K/3891200K available (14976K kernel code, 1776K rwdata, 4908K rodata, 3328K init, 658K bss, 114608K reserved, 262144K cma-reserved)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=6, Nodes=1
[    0.000000] rcu: Preemptible hierarchical RCU implementation.
[    0.000000] rcu:     RCU restricting CPUs from NR_CPUS=256 to nr_cpu_ids=6.
[    0.000000]  Trampoline variant of Tasks RCU enabled.
[    0.000000]  Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 25 jiffies.
[    0.000000] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=6
[    0.000000] NR_IRQS: 64, nr_irqs: 64, preallocated irqs: 0
[    0.000000] Root IRQ handler: gic_handle_irq
[    0.000000] GIC: Using split EOI/Deactivate mode
[    0.000000] rcu: srcu_init: Setting srcu_struct sizes based on contention.
[    0.000000] arch_timer: cp15 timer(s) running at 24.00MHz (phys).
[    0.000000] clocksource: arch_sys_counter: mask: 0xffffffffffffff max_cycles: 0x588fe9dc0, max_idle_ns: 440795202592 ns
[    0.000000] sched_clock: 56 bits at 24MHz, resolution 41ns, wraps every 4398046511097ns
[    0.000198] Console: colour dummy device 80x25
[    0.000207] printk: console [tty1] enabled
[    0.000302] Calibrating delay loop (skipped), value calculated using timer frequency.. 48.00 BogoMIPS (lpj=96000)
[    0.000311] pid_max: default: 32768 minimum: 301
[    0.000357] LSM: initializing lsm=capability,yama,integrity,selinux
[    0.000380] Yama: disabled by default; enable with sysctl kernel.yama.*
[    0.000398] SELinux:  Initializing.
[    0.000491] Mount-cache hash table entries: 8192 (order: 4, 65536 bytes, linear)
[    0.000505] Mountpoint-cache hash table entries: 8192 (order: 4, 65536 bytes, linear)
[    0.001968] cblist_init_generic: Setting adjustable number of callback queues.
[    0.001986] cblist_init_generic: Setting shift to 3 and lim to 1.
[    0.002036] cblist_init_generic: Setting shift to 3 and lim to 1.
[    0.002159] rcu: Hierarchical SRCU implementation.
[    0.002164] rcu:     Max phase no-delay instances is 1000.
[    0.004178] smp: Bringing up secondary CPUs ...
[    0.004748] Detected VIPT I-cache on CPU1
[    0.004861] CPU1: Booted secondary processor 0x0000000001 [0x410fd034]
[    0.005954] CPU features: detected: Spectre-v2
[    0.005967] CPU features: detected: Spectre-v4
[    0.005972] CPU features: detected: Spectre-BHB
[    0.005977] CPU features: detected: ARM erratum 858921
[    0.005983] Detected VIPT I-cache on CPU2
[    0.006069] arch_timer: Enabling local workaround for ARM erratum 858921
[    0.006084] arch_timer: CPU2: Trapping CNTVCT access
[    0.006093] CPU2: Booted secondary processor 0x0000000100 [0x410fd092]
[    0.006630] Detected VIPT I-cache on CPU3
[    0.006688] arch_timer: Enabling local workaround for ARM erratum 858921
[    0.006696] arch_timer: CPU3: Trapping CNTVCT access
[    0.006702] CPU3: Booted secondary processor 0x0000000101 [0x410fd092]
[    0.007229] Detected VIPT I-cache on CPU4
[    0.007290] arch_timer: Enabling local workaround for ARM erratum 858921
[    0.007298] arch_timer: CPU4: Trapping CNTVCT access
[    0.007304] CPU4: Booted secondary processor 0x0000000102 [0x410fd092]
[    0.007827] Detected VIPT I-cache on CPU5
[    0.007889] arch_timer: Enabling local workaround for ARM erratum 858921
[    0.007897] arch_timer: CPU5: Trapping CNTVCT access
[    0.007903] CPU5: Booted secondary processor 0x0000000103 [0x410fd092]
[    0.007962] smp: Brought up 1 node, 6 CPUs
[    0.007983] SMP: Total of 6 processors activated.
[    0.007987] CPU features: detected: 32-bit EL0 Support
[    0.007990] CPU features: detected: 32-bit EL1 Support
[    0.007994] CPU features: detected: CRC32 instructions
[    0.008064] CPU: All CPU(s) started at EL2
[    0.008080] alternatives: applying system-wide alternatives
[    0.009856] devtmpfs: initialized
[    0.016483] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7645041785100000 ns
[    0.016506] futex hash table entries: 2048 (order: 5, 131072 bytes, linear)
[    0.020598] pinctrl core: initialized pinctrl subsystem
[    0.021428] NET: Registered PF_NETLINK/PF_ROUTE protocol family
[    0.022275] DMA: preallocated 512 KiB GFP_KERNEL pool for atomic allocations
[    0.022600] DMA: preallocated 512 KiB GFP_KERNEL|GFP_DMA pool for atomic allocations
[    0.022875] DMA: preallocated 512 KiB GFP_KERNEL|GFP_DMA32 pool for atomic allocations
[    0.022906] audit: initializing netlink subsys (disabled)
[    0.023008] audit: type=2000 audit(0.020:1): state=initialized audit_enabled=0 res=1
[    0.023961] thermal_sys: Registered thermal governor 'fair_share'
[    0.023967] thermal_sys: Registered thermal governor 'bang_bang'
[    0.023971] thermal_sys: Registered thermal governor 'step_wise'
[    0.023974] thermal_sys: Registered thermal governor 'user_space'
[    0.024000] cpuidle: using governor menu
[    0.024093] hw-breakpoint: found 6 breakpoint and 4 watchpoint registers.
[    0.024181] ASID allocator initialised with 65536 entries
[    0.024977] Serial: AMBA PL011 UART driver
[    0.037557] platform ff900000.vpu: Fixing up cyclic dependency with ff600000.hdmi-tx
[    0.043542] platform hdmi-connector: Fixing up cyclic dependency with ff600000.hdmi-tx
[    0.044744] KASLR disabled due to lack of seed
[    0.045274] HugeTLB: registered 1.00 GiB page size, pre-allocated 0 pages
[    0.045280] HugeTLB: 16380 KiB vmemmap can be freed for a 1.00 GiB page
[    0.045286] HugeTLB: registered 32.0 MiB page size, pre-allocated 0 pages
[    0.045290] HugeTLB: 508 KiB vmemmap can be freed for a 32.0 MiB page
[    0.045294] HugeTLB: registered 2.00 MiB page size, pre-allocated 0 pages
[    0.045298] HugeTLB: 28 KiB vmemmap can be freed for a 2.00 MiB page
[    0.045302] HugeTLB: registered 64.0 KiB page size, pre-allocated 0 pages
[    0.045305] HugeTLB: 0 KiB vmemmap can be freed for a 64.0 KiB page
[    0.045945] cryptd: max_cpu_qlen set to 1000
[    0.111945] raid6: neonx8   gen()  2774 MB/s
[    0.179990] raid6: neonx4   gen()  2969 MB/s
[    0.248028] raid6: neonx2   gen()  2581 MB/s
[    0.316073] raid6: neonx1   gen()  2008 MB/s
[    0.384106] raid6: int64x8  gen()  1336 MB/s
[    0.452156] raid6: int64x4  gen()  1439 MB/s
[    0.520187] raid6: int64x2  gen()  1341 MB/s
[    0.588235] raid6: int64x1  gen()  1111 MB/s
[    0.588239] raid6: using algorithm neonx4 gen() 2969 MB/s
[    0.656268] raid6: .... xor() 2345 MB/s, rmw enabled
[    0.656272] raid6: using neon recovery algorithm
[    0.657127] iommu: Default domain type: Translated
[    0.657133] iommu: DMA domain TLB invalidation policy: strict mode
[    0.657310] SCSI subsystem initialized
[    0.657419] usbcore: registered new interface driver usbfs
[    0.657436] usbcore: registered new interface driver hub
[    0.657455] usbcore: registered new device driver usb
[    0.657700] pps_core: LinuxPPS API ver. 1 registered
[    0.657704] pps_core: Software ver. 5.3.6 - Copyright 2005-2007 Rodolfo Giometti <giometti@linux.it>
[    0.657715] PTP clock support registered
[    0.657823] EDAC MC: Ver: 3.0.0
[    0.658971] vgaarb: loaded
[    0.659173] clocksource: Switched to clocksource arch_sys_counter
[    0.659308] VFS: Disk quotas dquot_6.6.0
[    0.659331] VFS: Dquot-cache hash table entries: 512 (order 0, 4096 bytes)
[    0.663536] NET: Registered PF_INET protocol family
[    0.663738] IP idents hash table entries: 65536 (order: 7, 524288 bytes, linear)
[    0.666663] tcp_listen_portaddr_hash hash table entries: 2048 (order: 3, 32768 bytes, linear)
[    0.666692] Table-perturb hash table entries: 65536 (order: 6, 262144 bytes, linear)
[    0.666704] TCP established hash table entries: 32768 (order: 6, 262144 bytes, linear)
[    0.666847] TCP bind hash table entries: 32768 (order: 8, 1048576 bytes, linear)
[    0.667221] TCP: Hash tables configured (established 32768 bind 32768)
[    0.667323] UDP hash table entries: 2048 (order: 4, 65536 bytes, linear)
[    0.667365] UDP-Lite hash table entries: 2048 (order: 4, 65536 bytes, linear)
[    0.667522] NET: Registered PF_UNIX/PF_LOCAL protocol family
[    0.667554] PCI: CLS 0 bytes, default 64
[    0.667769] Trying to unpack rootfs image as initramfs...
[    0.675978] kvm [1]: IPA Size Limit: 40 bits
[    0.683312] kvm [1]: vgic interrupt IRQ9
[    0.683415] kvm [1]: Hyp mode initialized successfully
[    0.684474] Initialise system trusted keyrings
[    0.684630] workingset: timestamp_bits=44 max_order=20 bucket_order=0
[    0.684689] zbud: loaded
[    0.685139] JFS: nTxBlock = 8192, nTxLock = 65536
[    0.687689] SGI XFS with ACLs, security attributes, realtime, quota, no debug enabled
[    0.688122] ocfs2: Registered cluster interface o2cb
[    0.688198] OCFS2 User DLM kernel interface loaded
[    0.688922] gfs2: GFS2 installed
[    0.716499] xor: measuring software checksum speed
[    0.718879]    8regs           :  4145 MB/sec
[    0.721237]    32regs          :  4252 MB/sec
[    0.723925]    arm64_neon      :  3669 MB/sec
[    0.723929] xor: using function: 32regs (4252 MB/sec)
[    0.723938] Key type asymmetric registered
[    0.723942] Asymmetric key parser 'x509' registered
[    0.723982] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 245)
[    0.724155] io scheduler mq-deadline registered
[    0.724162] io scheduler kyber registered
[    0.724191] io scheduler bfq registered
[    0.724521] irq_meson_gpio: 100 to 8 gpio interrupt mux initialized
[    0.737306] soc soc0: Amlogic Meson G12B (S922X) Revision 29:a (40:2) Detected
[    0.741031] Serial: 8250/16550 driver, 4 ports, IRQ sharing enabled
[    0.742459] Serial: AMBA driver
[    0.742766] ff803000.serial: ttyAML0 at MMIO 0xff803000 (irq = 14, base_baud = 1500000) is a meson_uart
[    0.743214] ffd24000.serial: ttyAML1 at MMIO 0xffd24000 (irq = 15, base_baud = 1500000) is a meson_uart
[    0.747039] brd: module loaded
[    0.749845] loop: module loaded
[    0.751848] tun: Universal TUN/TAP device driver, 1.6
[    0.752028] usbcore: registered new interface driver cdc_ether
[    0.752054] usbcore: registered new interface driver cdc_subset
[    0.752277] VFIO - User Level meta-driver version: 0.3
[    0.753976] usbcore: registered new interface driver usb-storage
[    0.755124] mousedev: PS/2 mouse device common for all mice
[    0.755289] i2c_dev: i2c /dev entries driver
[    0.760037] cpufreq: cpufreq_online: CPU2: Running at unlisted initial frequency: 999999 KHz, changing to: 1000000 KHz
[    0.761317] sdhci: Secure Digital Host Controller Interface driver
[    0.761328] sdhci: Copyright(c) Pierre Ossman
[    0.761394] Synopsys Designware Multimedia Card Interface Driver
[    0.761691] sdhci-pltfm: SDHCI platform and OF driver helper
[    0.763407] ledtrig-cpu: registered to indicate activity on CPUs
[    0.763682] meson-sm: secure-monitor enabled
[    0.763835] hid: raw HID events driver (C) Jiri Kosina
[    0.764075] usbcore: registered new interface driver usbhid
[    0.764082] usbhid: USB HID core driver
[    0.765134] NET: Registered PF_INET6 protocol family
[    0.765533] Segment Routing with IPv6
[    0.765550] In-situ OAM (IOAM) with IPv6
[    0.765575] NET: Registered PF_PACKET protocol family
[    0.765581] NET: Registered PF_KEY protocol family
[    0.765623] 8021q: 802.1Q VLAN Support v1.8
[    0.765650] Key type dns_resolver registered
[    0.768711] registered taskstats version 1
[    0.768825] Loading compiled-in X.509 certificates
[    0.769007] zswap: loaded using pool lzo/zbud
[    0.772067] Key type .fscrypt registered
[    0.772076] Key type fscrypt-provisioning registered
[    0.772433] Btrfs loaded, crc32c=crc32c-generic, zoned=yes, fsverity=no
[    0.784246] meson-gx-mmc ffe05000.sd: Got CD GPIO
[    0.787923] dwc3-meson-g12a ffe09000.usb: USB2 ports: 2
[    0.787935] dwc3-meson-g12a ffe09000.usb: USB3 ports: 1
[    0.790599] dwc2 ff400000.usb: supply vusb_d not found, using dummy regulator
[    0.790653] dwc2 ff400000.usb: supply vusb_a not found, using dummy regulator
[    0.792882] dwc2 ff400000.usb: EPs: 7, dedicated fifos, 712 entries in SPRAM
[    0.793575] xhci-hcd xhci-hcd.0.auto: xHCI Host Controller
[    0.793590] xhci-hcd xhci-hcd.0.auto: new USB bus registered, assigned bus number 1
[    0.793666] xhci-hcd xhci-hcd.0.auto: hcc params 0x0228fe6c hci version 0x110 quirks 0x0000000000010010
[    0.793690] xhci-hcd xhci-hcd.0.auto: irq 21, io mem 0xff500000
[    0.793762] xhci-hcd xhci-hcd.0.auto: xHCI Host Controller
[    0.793768] xhci-hcd xhci-hcd.0.auto: new USB bus registered, assigned bus number 2
[    0.793772] xhci-hcd xhci-hcd.0.auto: Host supports USB 3.0 SuperSpeed
[    0.793826] usb usb1: New USB device found, idVendor=1d6b, idProduct=0002, bcdDevice= 6.02
[    0.793831] usb usb1: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    0.793834] usb usb1: Product: xHCI Host Controller
[    0.793837] usb usb1: Manufacturer: Linux 6.2.0-odroid-arm64 xhci-hcd
[    0.793840] usb usb1: SerialNumber: xhci-hcd.0.auto
[    0.794053] hub 1-0:1.0: USB hub found
[    0.794065] hub 1-0:1.0: 2 ports detected
[    0.794211] usb usb2: We don't know the algorithms for LPM for this host, disabling LPM.
[    0.794239] usb usb2: New USB device found, idVendor=1d6b, idProduct=0003, bcdDevice= 6.02
[    0.794243] usb usb2: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    0.794247] usb usb2: Product: xHCI Host Controller
[    0.794249] usb usb2: Manufacturer: Linux 6.2.0-odroid-arm64 xhci-hcd
[    0.794252] usb usb2: SerialNumber: xhci-hcd.0.auto
[    0.794391] hub 2-0:1.0: USB hub found
[    0.794400] hub 2-0:1.0: 1 port detected
[    0.794660] dwc3-meson-g12a ffe09000.usb: switching to Device Mode
[    0.795852] meson-gx-mmc ffe07000.mmc: allocated mmc-pwrseq
[    0.796174] clk: Not disabling unused clocks
[    0.811398] genirq: Setting trigger mode 3 for irq 24 failed (meson_gpio_irq_set_type+0x0/0x60)
[    0.893760] Freeing initrd memory: 9368K
[    0.899647] Freeing unused kernel memory: 3328K
[    0.933296] mmc0: new HS200 MMC card at address 0001
[    0.933703] mmcblk0: mmc0:0001 BJTD4R 29.1 GiB
[    0.934709]  mmcblk0: p1 p2 p3 < p5 >
[    0.935270] mmcblk0boot0: mmc0:0001 BJTD4R 4.00 MiB
[    0.936128] mmcblk0boot1: mmc0:0001 BJTD4R 4.00 MiB
[    0.936676] mmcblk0rpmb: mmc0:0001 BJTD4R 4.00 MiB, chardev (242:0)
[    0.947335] Run /init as init process
[    0.947347]   with arguments:
[    0.947350]     /init
[    0.947353]   with environment:
[    0.947354]     HOME=/
[    0.947356]     TERM=linux
[    1.087213] usb 1-1: new high-speed USB device number 2 using xhci-hcd
[    1.089111] meson8b-dwmac ff3f0000.ethernet: IRQ eth_wake_irq not found
[    1.089124] meson8b-dwmac ff3f0000.ethernet: IRQ eth_lpi not found
[    1.089218] meson8b-dwmac ff3f0000.ethernet: PTP uses main clock
[    1.089655] meson8b-dwmac ff3f0000.ethernet: User ID: 0x11, Synopsys ID: 0x37
[    1.089671] meson8b-dwmac ff3f0000.ethernet:         DWMAC1000
[    1.089675] meson8b-dwmac ff3f0000.ethernet: DMA HW capability register supported
[    1.089678] meson8b-dwmac ff3f0000.ethernet: RX Checksum Offload Engine supported
[    1.089680] meson8b-dwmac ff3f0000.ethernet: COE Type 2
[    1.089683] meson8b-dwmac ff3f0000.ethernet: TX Checksum insertion supported
[    1.089685] meson8b-dwmac ff3f0000.ethernet: Wake-Up On Lan supported
[    1.089745] meson8b-dwmac ff3f0000.ethernet: Normal descriptors
[    1.089749] meson8b-dwmac ff3f0000.ethernet: Ring mode enabled
[    1.089751] meson8b-dwmac ff3f0000.ethernet: Enable RX Mitigation via HW Watchdog Timer
[    1.089756] meson8b-dwmac ff3f0000.ethernet: device MAC address 00:1e:06:42:44:b6
[    1.109685] rtc-pcf8563 0-0051: registered as rtc0
[    1.110997] rtc-pcf8563 0-0051: setting system clock to 2024-12-24T20:08:15 UTC (1735070895)
[    1.237515] usb 1-1: New USB device found, idVendor=05e3, idProduct=0610, bcdDevice=93.04
[    1.237529] usb 1-1: New USB device strings: Mfr=1, Product=2, SerialNumber=0
[    1.237532] usb 1-1: Product: USB2.1 Hub
[    1.237535] usb 1-1: Manufacturer: GenesysLogic
[    1.273983] hub 1-1:1.0: USB hub found
[    1.274202] hub 1-1:1.0: 4 ports detected
[    1.363304] usb 2-1: new SuperSpeed USB device number 2 using xhci-hcd
[    1.385773] usb 2-1: New USB device found, idVendor=05e3, idProduct=0620, bcdDevice=93.04
[    1.385783] usb 2-1: New USB device strings: Mfr=1, Product=2, SerialNumber=0
[    1.385788] usb 2-1: Product: USB3.1 Hub
[    1.385792] usb 2-1: Manufacturer: GenesysLogic
[    1.418040] hub 2-1:1.0: USB hub found
[    1.418340] hub 2-1:1.0: 4 ports detected
[    1.453745] PM: Image not found (code -22)
[    1.946026] EXT4-fs (mmcblk0p2): mounted filesystem 4d5a219f-f8ab-4770-ad3a-fa45dfd7b4f1 with ordered data mode. Quota mode: none.
[    2.114333] systemd[1]: systemd 247.3-7+deb11u6 running in system mode. (+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +ZSTD +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=unified)
[    2.114570] systemd[1]: Detected architecture arm64.
[    2.115052] systemd[1]: Set hostname to <HyperHDR>.
[    2.304702] systemd[1]: Queued start job for default target Graphical Interface.
[    3.415270] usb 2-1.3: new SuperSpeed USB device number 3 using xhci-hcd
[    3.435595] usb 2-1.3: LPM exit latency is zeroed, disabling LPM.
[    3.438062] usb 2-1.3: config 1 has an invalid interface number: 7 but max is 4
[    3.438068] usb 2-1.3: config 1 has no interface number 4
[    3.438085] usb 2-1.3: Int endpoint with wBytesPerInterval of 1024 in config 1 interface 7 altsetting 0 ep 135: setting to 512
[    3.438586] usb 2-1.3: New USB device found, idVendor=32ed, idProduct=3200, bcdDevice= 1.00
[    3.438592] usb 2-1.3: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[    3.438597] usb 2-1.3: Product: ezcap Game Link RAW
[    3.438601] usb 2-1.3: Manufacturer: ezcap
[    3.438604] usb 2-1.3: SerialNumber: 00000001
[    3.516175] hid-generic 0003:32ED:3200.0001: hiddev96,hidraw0: USB HID v1.11 Device [ezcap ezcap Game Link RAW] on usb-xhci-hcd.0.auto-1.3/input7
[    5.479182] random: crng init done
[    5.495949] systemd[1]: Created slice system-getty.slice.
[    5.496519] systemd[1]: Created slice system-hyperhdr.slice.
[    5.496902] systemd[1]: Created slice system-modprobe.slice.
[    5.497294] systemd[1]: Created slice system-systemd\x2dfsck.slice.
[    5.497631] systemd[1]: Created slice User and Session Slice.
[    5.497765] systemd[1]: Started Dispatch Password Requests to Console Directory Watch.
[    5.497854] systemd[1]: Started Forward Password Requests to Wall Directory Watch.
[    5.498167] systemd[1]: Set up automount Arbitrary Executable File Formats File System Automount Point.
[    5.498244] systemd[1]: Reached target Local Encrypted Volumes.
[    5.498310] systemd[1]: Reached target Paths.
[    5.498343] systemd[1]: Reached target Remote File Systems.
[    5.498366] systemd[1]: Reached target Slices.
[    5.498642] systemd[1]: Listening on Syslog Socket.
[    5.498810] systemd[1]: Listening on fsck to fsckd communication Socket.
[    5.498913] systemd[1]: Listening on initctl Compatibility Named Pipe.
[    5.499197] systemd[1]: Listening on Journal Audit Socket.
[    5.499350] systemd[1]: Listening on Journal Socket (/dev/log).
[    5.499551] systemd[1]: Listening on Journal Socket.
[    5.500094] systemd[1]: Listening on udev Control Socket.
[    5.500235] systemd[1]: Listening on udev Kernel Socket.
[    5.501700] systemd[1]: Mounting Huge Pages File System...
[    5.503368] systemd[1]: Mounting POSIX Message Queue File System...
[    5.505084] systemd[1]: Mounting Kernel Debug File System...
[    5.505356] systemd[1]: Condition check resulted in Kernel Trace File System being skipped.
[    5.507334] systemd[1]: Starting Set the console keyboard layout...
[    5.509339] systemd[1]: Starting Create list of static device nodes for the current kernel...
[    5.511295] systemd[1]: Starting Load Kernel Module configfs...
[    5.513439] systemd[1]: Starting Load Kernel Module drm...
[    5.515582] systemd[1]: Starting Load Kernel Module fuse...
[    5.517981] systemd[1]: Condition check resulted in Set Up Additional Binary Formats being skipped.
[    5.518128] systemd[1]: Condition check resulted in File System Check on Root Device being skipped.
[    5.521291] systemd[1]: Starting Journal Service...
[    5.525626] systemd[1]: Starting Load Kernel Modules...
[    5.527560] systemd[1]: Starting Remount Root and Kernel File Systems...
[    5.529531] systemd[1]: Starting Coldplug All udev Devices...
[    5.533833] systemd[1]: Mounted Huge Pages File System.
[    5.534165] systemd[1]: Mounted POSIX Message Queue File System.
[    5.534446] systemd[1]: Mounted Kernel Debug File System.
[    5.535406] systemd[1]: Finished Create list of static device nodes for the current kernel.
[    5.536276] systemd[1]: modprobe@configfs.service: Succeeded.
[    5.536815] systemd[1]: Finished Load Kernel Module configfs.
[    5.539547] systemd[1]: Mounting Kernel Configuration File System...
[    5.542763] systemd[1]: modprobe@drm.service: Succeeded.
[    5.543483] systemd[1]: Finished Load Kernel Module drm.
[    5.544549] fuse: init (API version 7.38)
[    5.546575] systemd[1]: modprobe@fuse.service: Succeeded.
[    5.547233] systemd[1]: Finished Load Kernel Module fuse.
[    5.548136] systemd[1]: Finished Load Kernel Modules.
[    5.548456] systemd[1]: Mounted Kernel Configuration File System.
[    5.548935] EXT4-fs (mmcblk0p2): re-mounted 4d5a219f-f8ab-4770-ad3a-fa45dfd7b4f1. Quota mode: none.
[    5.550396] systemd[1]: Mounting FUSE Control File System...
[    5.552369] systemd[1]: Starting Apply Kernel Variables...
[    5.555995] systemd[1]: Finished Remount Root and Kernel File Systems.
[    5.556433] systemd[1]: Mounted FUSE Control File System.
[    5.557556] systemd[1]: Condition check resulted in Rebuild Hardware Database being skipped.
[    5.557673] systemd[1]: Condition check resulted in Platform Persistent Storage Archival being skipped.
[    5.559556] systemd[1]: Starting Load/Save Random Seed...
[    5.561644] systemd[1]: Starting Create System Users...
[    5.570591] systemd[1]: Finished Apply Kernel Variables.
[    5.583667] systemd[1]: Finished Load/Save Random Seed.
[    5.583993] systemd[1]: Condition check resulted in First Boot Complete being skipped.
[    5.594048] systemd[1]: Finished Create System Users.
[    5.595887] systemd[1]: Starting Create Static Device Nodes in /dev...
[    5.613731] systemd[1]: Started Journal Service.
[    5.626141] systemd-journald[1003]: Received client request to flush runtime journal.
[    5.816353] gpio-fan gpio-fan: GPIO fan initialized
[    5.882799] mc: Linux media interface: v0.10
[    5.897129] videodev: Linux video capture interface: v2.00
[    5.925853] Registered IR keymap rc-odroid
[    5.933068] IR NEC protocol handler initialized
[    5.942146] meson-drm ff900000.vpu: Queued 1 outputs on vpu
[    5.948552] meson-drm ff900000.vpu: CVBS Output connector not available
[    5.964558] meson_vdec: module is from the staging directory, the quality is unknown, you have been warned.
[    5.965847] rc rc0: meson-ir as /devices/platform/soc/ff800000.bus/ff808000.ir/rc/rc0
[    5.966142] rc rc0: lirc_dev: driver meson-ir registered at minor = 0, raw IR receiver, no transmitter
[    5.968354] input: meson-ir as /devices/platform/soc/ff800000.bus/ff808000.ir/rc/rc0/input0
[    5.969424] meson-ir ff808000.ir: receiver initialized
[    5.987319] meson-dw-hdmi ff600000.hdmi-tx: Detected HDMI TX controller v2.01a with HDCP (meson_dw_hdmi_phy)
[    6.012469] panfrost ffe40000.gpu: clock rate = 24000000
[    6.012553] panfrost ffe40000.gpu: error -ENODEV: _opp_set_regulators: no regulator (mali) found
[    6.022457] meson-dw-hdmi ff600000.hdmi-tx: registered DesignWare HDMI I2C bus driver
[    6.034320] meson-drm ff900000.vpu: bound ff600000.hdmi-tx (ops meson_dw_hdmi_ops [meson_dw_hdmi])
[    6.039607] panfrost ffe40000.gpu: mali-g52 id 0x7212 major 0x0 minor 0x0 status 0x0
[    6.039624] panfrost ffe40000.gpu: features: 00000000,00000cf7, issues: 00000000,00000400
[    6.039629] panfrost ffe40000.gpu: Features: L2:0x07110206 Shader:0x00000000 Tiler:0x00000809 Mem:0x1 MMU:0x00002830 AS:0xff JS:0x7
[    6.039635] panfrost ffe40000.gpu: shader_present=0x3 l2_present=0x1
[    6.047224] [drm] Initialized meson 1.0.0 20161109 for ff900000.vpu on minor 0
[    6.063690] [drm] Initialized panfrost 1.2.0 20180908 for ffe40000.gpu on minor 1
[    6.073167] uvcvideo 2-1.3:1.1: Unknown video format e436eb7e-524f-11ce-9f53-0020af0ba770
[    6.073196] usb 2-1.3: Found UVC 1.10 device ezcap Game Link RAW (32ed:3200)
[    6.094318] Adding 998396k swap on /dev/mmcblk0p5.  Priority:-2 extents:1 across:998396k SSFS
[    6.098761] input: ezcap Game Link RAW: ezcap Game as /devices/platform/soc/ffe09000.usb/ff500000.usb/xhci-hcd.0.auto/usb2/2-1/2-1.3/2-1.3:1.0/input/input1
[    6.098968] usbcore: registered new interface driver uvcvideo
[    6.128635] usbcore: registered new interface driver snd-usb-audio
[    6.251984] Console: switching to colour frame buffer device 128x48
[    6.269904] meson-drm ff900000.vpu: [drm] fb0: mesondrmfb frame buffer device
[    6.613272] ext2 filesystem being mounted at /boot supports timestamps until 2038 (0x7fffffff)
[    6.696203] meson8b-dwmac ff3f0000.ethernet eth0: Register MEM_TYPE_PAGE_POOL RxQ-0
[    6.871338] meson8b-dwmac ff3f0000.ethernet eth0: PHY [mdio_mux-0.0:00] driver [RTL8211F Gigabit Ethernet] (irq=POLL)
[    6.879415] meson8b-dwmac ff3f0000.ethernet eth0: No Safety Features support found
[    6.879434] meson8b-dwmac ff3f0000.ethernet eth0: PTP not supported by HW
[    6.879669] meson8b-dwmac ff3f0000.ethernet eth0: configuring for phy/rgmii link mode
[   11.009859] meson8b-dwmac ff3f0000.ethernet eth0: Link is Up - 1Gbps/Full - flow control rx/tx
[   11.009893] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
`
